### PR TITLE
Update to wgpu 22.1.0

### PR DESCRIPTION
**Pull request by waywardmonkeys**

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
This is blocked on https://github.com/Wumpf/wgpu-profiler/pull/75

That in turn is very likely blocked on getting a 22.0.1 release of `wgpu`.

Opening this here now to track that it is in progress and what's blocking it as well as call out a couple of discussion points.

1) In `vello_shaders`, we have a feature `msl` which could be changed to only happen on Apple targets by using `msl-out-if-target-apple` instead with Naga 22.  I don't know what the upstream users of this feature want.
2) We could start to enable the work from #524 to improve startup time when using Vulkan.
3) ???

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ This release has an [MSRV][] of 1.75.
 
 ### Changed
 
+- Updated `wgpu` to 22.1.0. ([#635] by [@waywardmonkeys])
```
Comment (DJMcNab) [inline CHANGELOG.md:22]: We don't have a consistent policy here. Is it worth marking this as a breaking change?

File: vello/src/wgpu_engine.rs [hunk 2]
```diff
@@ -678,6 +678,7 @@ impl WgpuEngine {
                 zero_initialize_workgroup_memory: false,
                 ..Default::default()
             },
+            cache: None,
```
Comment (DJMcNab) [inline vello/src/wgpu_engine.rs:681]:
```suggestion
            // TODO: Support providing a cache here
            cache: None,
```

**Review by DJMcNab — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (waywardmonkeys) [reply CHANGELOG.md:22]: I think it should be for now ... if we don't, it would cause some problems for users of `bevy_vello`.

Comment (DJMcNab) [reply CHANGELOG.md:22]: I wasn't suggesting we don't treat it as a breaking change. I mean to ask whether we should call it out as a breaking change in the changelog

Comment (waywardmonkeys) [reply CHANGELOG.md:22]: Oh, sorry. Doing too many things at once. How should we do that? I'm okay with doing so here.

Comment (waywardmonkeys) [reply CHANGELOG.md]: I did this like your own entry now. Will merge soon.

**Pull request merged**